course_inventory/published_date.py

```diff
@@ -75,40 +106,113 @@ def published_date_resp_parsing(self, response):
                 published_date_found = True
                 self.published_course_date.update({course_id: event['created_at']})
                 logger.info(f"Published Date {event['created_at']} for course {course_id}")
+                if course_id in self.published_date_retry_bucket:
+                    logger.info(f"Going to remove {course_id} from retry list {self.published_date_retry_bucket}")
+                    self.published_date_retry_bucket.pop(course_id)
+                    logger.info(f"Removed {course_id} removed from retry list {self.published_date_retry_bucket}")
                 break
         if not published_date_found:
             self.get_next_page_url(response)
 
-        seconds = time.time() - start_time
-        str_time = time.strftime("%H:%M:%S", time.gmtime(seconds))
-        logger.debug(f"Parsing the published date took {str_time} ")
         return
 
-    def get_published_course_date(self, course_ids, next_page_links=None):
-        logger.info("Starting of get_published_course_date call")
-        with FuturesSession(max_workers=self.num_workers) as session:
+    def retry_logic_with_error(self, course_id, url) -> None:
+        if course_id in self.published_date_retry_bucket:
+            retry_count = self.published_date_retry_bucket[course_id]['count']
+            if retry_count > self.retry_attempts - 1:
```

Comment:
Nit-pick: I think you could make this `if retry_count >= self.retry_attempts` if you didn't want to subtract. You might not even need the `>` sign...

Reply:
Will make the change

Reply:
Done